services/mlmodel/tflite_cpu/tflite_cpu_test.go

```diff
@@ -0,0 +1,113 @@
+package tflitecpu
+
+import (
+	"context"
+	"testing"
+
+	"github.com/nfnt/resize"
+	"go.viam.com/test"
+	"go.viam.com/utils/artifact"
+
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/services/mlmodel"
+)
+
+func TestNewTFLiteCPUModel(t *testing.T) {
+	// Setup
+	ctx := context.Background()
+	modelLoc := artifact.MustPath("vision/tflite/effdet0.tflite")
+	modelLoc2 := artifact.MustPath("vision/tflite/effnet0.tflite")
+	emptyCfg := TFLiteConfig{} // empty config
+	cfg := TFLiteConfig{       // detector config
+		ModelPath:  modelLoc,
+		NumThreads: 2,
+	}
+	cfg2 := TFLiteConfig{ // classifier config
+		ModelPath:  modelLoc2,
+		NumThreads: 2,
+	}
+
+	// Test that empty config gives error about loading model
+	emptyGot, err := CreateTFLiteCPUModel(ctx, &emptyCfg)
+	test.That(t, emptyGot, test.ShouldResemble, &Model{})
+	test.That(t, err.Error(), test.ShouldContainSubstring, "could not add model")
+	test.That(t, cfg, test.ShouldNotBeNil)
+
+	// Test that a detector would give the expected output on the dog image
+	// Creating the model should populate model and attrs, but not metadata
+	got, err := CreateTFLiteCPUModel(ctx, &cfg)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, got.model, test.ShouldNotBeNil)
+	test.That(t, got.attrs, test.ShouldNotBeNil)
+	test.That(t, got.metadata, test.ShouldBeNil)
+
+	// Test that the Metadata() works on detector
+	gotMD, err := got.Metadata(ctx)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, gotMD, test.ShouldNotBeNil)
+	test.That(t, got.metadata, test.ShouldNotBeNil)
+
+	test.That(t, gotMD.Inputs[0].Name, test.ShouldResemble, "image")
+	test.That(t, gotMD.Outputs[0].Name, test.ShouldResemble, "location")
+	test.That(t, gotMD.Outputs[1].Name, test.ShouldResemble, "category")
+	test.That(t, gotMD.Outputs[2].Name, test.ShouldResemble, "score")
+	test.That(t, gotMD.Inputs[0].DataType, test.ShouldResemble, "uint8")
+	test.That(t, gotMD.Outputs[0].DataType, test.ShouldResemble, "float32")
+	test.That(t, gotMD.Outputs[1].AssociatedFiles[0].Name, test.ShouldResemble, "labelmap.txt")
+
+	// Test that the Infer() works on detector
+	pic, err := rimage.NewImageFromFile(artifact.MustPath("vision/tflite/dogscute.jpeg"))
+	test.That(t, err, test.ShouldBeNil)
+	resized := resize.Resize(uint(got.model.Info.InputWidth), uint(got.model.Info.InputHeight), pic, resize.Bilinear)
```

Comment:
Didn't realize this before but it's kinda uncool to me that the input size for the model is separate from the rest of the metadata (especially since mlmodel service users will have to handle the resize). I can easily grab it from model.Info.InputSize ([]int) and funnel it into the metadata struct.  Only problem is it'll be a proto/scope change.   (We could also swap out NDim (int) from the TensorInfo and replace it with TensorShape ([]int) even though the info will only be important for the input tensor.)